gnd/src/main/java/com/google/android/gnd/ui/offlinearea/selector/OfflineAreaSelectorViewModel.java

```diff
@@ -16,27 +16,51 @@
 
 package com.google.android.gnd.ui.offlinearea.selector;
 
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
 import com.google.android.gms.maps.model.LatLngBounds;
 import com.google.android.gnd.repository.OfflineAreaRepository;
 import com.google.android.gnd.ui.common.AbstractViewModel;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.subjects.PublishSubject;
 import javax.inject.Inject;
 import timber.log.Timber;
 
 public class OfflineAreaSelectorViewModel extends AbstractViewModel {
 
+  private final LiveData<DownloadTrigger> downloads;
```

Comment:
I believe in this case the toast will re-appear every time to view is restored (e.g. after configuration change). To ensure they're single use, you can wrap DownloadTrigger in our custom class  `Event<>`.

Reply:
Done